src/index.js

```diff
@@ -109,12 +115,22 @@ const decorator = ({ declareQueries, declareCommands, declareConnect }) => (Comp
   ]);
 
   @composedDecorators
-  @skinnable(contains(Component))
+  @skinnable(contains(loadingDecorator(Component)))
   @pure
   @props(propsTypes)
   class ContainerFactoryWrapper extends React.Component { // eslint-disable-line react/no-multi-comp
+
     static displayName = displayName(Component, 'Container');
-    getLocals = getLocals;
+
+    getLocals(props) {
+      const { readyState } = props;
+      if (isReady(props)) {
+        return { ...getLocals(props), ...stripUndef({ readyState }) };
+      } else {
+        return stripUndef({ readyState });
```

Comment:
**minor and subjective** I prefer APIs that return a JSON with always the same keys: I would not strip the `undefined`s (which IMO should only be `error`) and instead set it to `null`